### PR TITLE
*: add `SnapshotIterReverse` and make `iterReverse` supports `lowerBound`

### PR DESCRIPTION
**Pull request by Defined2014**
For https://github.com/pingcap/tidb/issues/45315, add `lowerBound` param for `iterReverse` function.

### CODE REVIEW

**Review by tiancaiamao — APPROVED** [review]

**Comment by Defined2014** [comment]
cc @disksing

File: integration_tests/go.mod [hunk 1]
```diff
@@ -113,5 +113,7 @@ require (
 
 replace (
 	github.com/go-ldap/ldap/v3 => github.com/YangKeao/ldap/v3 v3.4.5-0.20230421065457-369a3bab1117
+	github.com/pingcap/tidb => github.com/defined2014/tidb v1.1.0-beta.0.20230712071811-87a5c15e9c7d
```
Comment (HuSharp) [inline integration_tests/go.mod:116]: Do you have any plans to update it?

Comment (Defined2014) [reply integration_tests/go.mod:116]: Yes, it think so. You could update to the master.

**Pull request merged**